http/http_errors.ts

```diff
@@ -46,160 +49,7 @@
-      ? isClientErrorStatus(this.status)
-      : options.expose;
-    if (options?.headers) {
-      this.#headers = new Headers(options.headers);
-    }
-  }
-  /** A flag to indicate if the internals of the error, like the stack, should
-   * be exposed to a client, or if they are "private" and should not be leaked.
-   * By default, all client errors are `true` and all server errors are
-   * `false`. */
-  get expose(): boolean {
-    return this.#expose;
-  }
-  /** The optional headers object that is set on the error. */
-  get headers(): Headers | undefined {
-    return this.#headers;
-  }
-  /** The error status that is set on the error. */
-  get status(): ErrorStatus {
-    return this.#status;
-  }
-}
-
-function createHttpErrorConstructor(status: ErrorStatus): typeof HttpError {
-  const name = `${Status[status]}Error`;
-  const ErrorCtor = class extends HttpError {
-    constructor(
-      message = STATUS_TEXT[status],
-      options?: HttpErrorOptions,
-    ) {
-      super(message, options);
-      Object.defineProperty(this, "name", {
-        configurable: true,
-        enumerable: false,
-        value: name,
-        writable: true,
-      });
-    }
-
-    override get status() {
-      return status;
-    }
-  };
-  return ErrorCtor;
-}
-
-/**
- * A namespace that contains each error constructor. Each error extends
- * `HTTPError` and provides `.status` and `.expose` properties, where the
- * `.status` will be an error `Status` value and `.expose` indicates if
- * information, like a stack trace, should be shared in the response.
- *
- * By default, `.expose` is set to false in server errors, and true for client
- * errors.
- *
- * @example
- * ```ts
- * import { errors } from "https://deno.land/std@$STD_VERSION/http/http_errors.ts";
- *
- * throw new errors.InternalServerError("Ooops!");
- * ```
- */
-export const errors: Record<ErrorStatusKeys, typeof HttpError> = {} as Record<
-  ErrorStatusKeys,
-  typeof HttpError
->;
-
-for (const [key, value] of Object.entries(ERROR_STATUS_MAP)) {
-  errors[key as ErrorStatusKeys] = createHttpErrorConstructor(value);
-}
-
 /**
- * A factory function which provides a way to create errors. It takes up to 3
- * arguments, the error `Status`, an message, which defaults to the status text
- * and error options, which includes the `expose` property to set the `.expose`
- * value on the error.
+ * @deprecated (will be removed after 0.210.0) Import from `std/http/unstable_errors.ts` instead.
  */
-export function createHttpError(
-  status: ErrorStatus = Status.InternalServerError,
-  message?: string,
-  options?: HttpErrorOptions,
-): HttpError {
-  return new errors[Status[status] as ErrorStatusKeys](message, options);
-}
-
-/** A type guard that determines if the value is an HttpError or not. */
-export function isHttpError(value: unknown): value is HttpError {
-  return value instanceof HttpError;
-}
+export * from "./unstable_errors.ts";
```

Comment:
Are the original JSDocs retained or does the deprecation notice overwrite them all? Same question elsewhere.

Reply:
I'm not sure... I suspect it's up to the editor (or maybe it's spec'd out and I'm completely wrong)? My editor LSP is kind of borked so I might not be the best person to ask.

Reply:
I’m sure it completely overwrites the rest of the JSDocs. I recommend exporting each function/class/whatever with its respective JSDoc with the prepended deprecation notice.

Reply:
Will do.